File: TestSample/cpp/src/Person.cpp
#include "Person.h"

namespace helloworld {

Person::Person(std::string name, std::string language)
    : name_(std::move(name)), language_(std::move(language)) {}

std::string Person::hello() const {
    return "[" + language_ + "] " + name_ + ": " + speak();
}

}

File: TestSample/typescript/src/Person.ts
export abstract class Person {
    readonly name: string;
    readonly language: string;

    constructor(name: string, language: string) {
        this.name = name;
        this.language = language;
    }

    abstract speak(): string;

    hello(): string {
        return `[${this.language}] ${this.name}: ${this.speak()}`;
    }
}
